jerry-core/jerryscript-port.h

```diff
@@ -150,6 +150,25 @@ void jerry_port_jobqueue_enqueue (jerry_job_handler_t handler, void *job_p);
 
 #endif /* !CONFIG_DISABLE_ES2015_PROMISE_BUILTIN */
 
+#ifdef JERRY_ENABLE_EXTERNAL_CONTEXT
+#define JERRY_PORT_ENABLE_EXTERNAL_CONTEXT
+/**
+ * Description of jerry instance, which contains the jerry context, heap and lcache.
+ */
+typedef struct
+{
+  void *context_p; /**< points to the external context */
+  void *heap_p; /**< points to the external heap */
+  void *lcache_p; /**< points to the external lcache */
+  uint32_t heap_size; /**< the total size of heap */
+  uint32_t lcache_row_count; /**< the row count of lcache hash table */
+  uint32_t lcache_row_length; /**< the row length of lcache hash table */
+} jerry_instance_t;
```

Comment:
Since this is a critical part of the engine, yes, we want to change the API if something changes. And yes, port maintainers must follow those changes. These changes will be rare though.